app/models/manageiq/providers/base_manager/metrics_capture.rb

```diff
@@ -175,18 +175,41 @@ def create_rollup_task_for_cluster(ems_cluster, hosts)
     end
   end
 
+  # number of records to send at a time for a single capture message
+  # @return [Numeric] 0 for no batching.  (defaults to 0)
+  def batch_size
+    @batch_size ||= ::Settings.dig(:ems, "ems_#{ems.provider_name.underscore}", :capture_batch_size).to_i
```

Comment:
Is there a better way of getting this settings key @agrare - feels like this should be an Ask Don't Tell situation where we ask the ems for it's settings key?

Reply:
Why have a setting?  Feels like there should be a method in the provider, and if that method chooses to have a  setting, then it can (and then you also don't need any interpolation here).

I generally have an aversion to settings unless there's a good reason a user might change it.

Reply:
When looking through `settings.yml`, this looked to be the suggested way.

I did have `::Settings.ems["ems_#{ems.provider_name.underscore}"]&.capture_batch_size` but some tests complained/blew up. (I assume because the full settings is not available in tests)

- configure feature on/off by ems (via code or settings)
- configure batching value by ems (via constant or settings)

The way I turned this on/off changed quite a bit while developing this.

Reply:
https://github.com/ManageIQ/manageiq-providers-vmware/pull/834

Reply:
Yeah it is too bad the setting isn't `Settings.dig(:ems, ems.emstype)` like it is for `ems_refresh`

If we don't already we could add an `ems_settings` method similar to the `worker_settings`

Reply:
@agrare I can change it to that, I just tried to follow what was out there.

Reply:
Not necessary to do here, moving to ems_settings can be in its own PR